packages/components/src/color-picker/index.js

```diff
@@ -106,140 +113,134 @@ const dataToColors = ( oldColors, { source, valueKey, value } ) => {
 		source,
 		...{ ...oldColors[ source ], ...{ [ valueKey ]: value } },
 	};
-};
-
-export default class ColorPicker extends Component {
-	constructor( { color = '0071a1' } ) {
-		super( ...arguments );
-		const colors = colorToState( color );
-		this.state = {
-			...colors,
-			draftHex: toLowerCase( colors.hex ),
-			draftRgb: colors.rgb,
-			draftHsl: colors.hsl,
-		};
-		this.commitValues = this.commitValues.bind( this );
-		this.setDraftValues = this.setDraftValues.bind( this );
-		this.resetDraftValues = this.resetDraftValues.bind( this );
-		this.handleInputChange = this.handleInputChange.bind( this );
-	}
-
-	commitValues( data ) {
-		const { oldHue, onChangeComplete = noop } = this.props;
+}
 
+export default function ColorPicker( {
+	color = '0071a1',
+	className,
+	disableAlpha,
+	oldHue,
+	onChangeComplete = noop,
+} ) {
+	const initialColor = colorToState( color );
+	const [ colors, setColors ] = useState( {
+		...initialColor,
+		draftHex: toLowerCase( initialColor.hex ),
+		draftRgb: initialColor.rgb,
+		draftHsl: initialColor.hsl,
+	} );
+
+	const debouncedOnChangeComplete = useDebounce( onChangeComplete, 100 );
+
+	const { hex, hsl, hsv, rgb, draftHex, draftHsl, draftRgb } = colors;
+
+	const classes = classnames( className, {
+		'components-color-picker': true,
+		'is-alpha-disabled': disableAlpha,
+		'is-alpha-enabled': ! disableAlpha,
+	} );
+
+	function commitValues( data ) {
 		if ( isValidColor( data ) ) {
-			const colors = colorToState( data, data.h || oldHue );
-			this.setState(
-				{
-					...colors,
-					draftHex: toLowerCase( colors.hex ),
-					draftHsl: colors.hsl,
-					draftRgb: colors.rgb,
-				},
-				debounce( partial( onChangeComplete, colors ), 100 )
```

Comment:
Tbh, I'm not sure how this work... If `onChangeComplete` is a new function on every render, the debouncing should fail?

Reply:
At least it should fail between re-renders. Not sure how many times this component re-renders.

Reply:
I think i should use a useCallback hook. Right ?!

Reply:
Yes, it would avoid creating new functions on every render, which is good for `debounce` and `throttle`. The only problem is that it should be done by the consumer of `ColorPicker`, since only the consumer knows if there are any dependencies.

Reply:
Maybe it's ok to leave it as it is now and only add `useCallback` for internal function over which `ColorPicker` has control.

Reply:
a added `useCallback` in 716869d6cb2724ce2f136be8fec380cd409ad825 inside the `ColorPicker`.

Reply:
This is useless if parent components return new functions on every render though: https://github.com/WordPress/gutenberg/search?q=onChangeComplete.

Not sure what's the best way around it.